pkg/kubelet/kubelet_node_status.go

```diff
@@ -579,7 +579,12 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 	}
 
 	node, changed := kl.updateNode(ctx, originalNode)
-	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency
+	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency+kl.delayAfter1stNodeStatusUpdate
+
+	// Only reset nodeStatusReportWaitDurationAfterInitialUpdate to 0 when it is used once.
+	if !kl.lastStatusReportTime.IsZero() && shouldPatchNodeStatus && !changed {
```

Comment:
why do we need `!changed` here?

Reply:
I have re-organize the code and hopefully it's easier to reason about.
```
	shouldPatchNodeStatus := changed || passStatusReportInterval
	if changed {
		// generate a random delay
	}

	if passStatusReportInterval {
		// reset nodeStatusReportWaitDurationAfterInitialUpdate to 0 when we use this random interval once.
	}

```